prototypes_exercise/prototypes.js

```diff
@@ -0,0 +1,78 @@
+function Person(
+	firstName, lastName, favoriteColor, favoriteNumber, favoriteFoods =[]
+	)
+{	//"Elie", "Schoppik", "purple", 34
+	this.firstName = firstName;
+	this.lastName = lastName;
+	this.favoriteColor = favoriteColor;
+	this.favoriteNumber = favoriteNumber;
+	this.favoriteFoods = favoriteFoods;
+	this.fullName = function(){ return `${this.firstName} ${this.lastName}`;};
+	this.family = [];
+}
+
+Person.prototype.toString = function()
+	{ return `${this.fullName()}, Favorite Color: ${this.favoriteColor}, Favorite Number: ${this.favoriteNumber}` };
+Person.prototype.addToFamily = function(fm){
+	if(this.family.includes(fm)) return this.family.length;
+	if(fm instanceof Person === false) return this.family.length;
+
+	this.family.push(fm);
+
+	return this.family.length;
+};
+
+Array.prototype.map = function(fn, keepUndefined = false){
```

Comment:
the second parameter to push is used to set the keyword this in the callback function.  So it would be something like:

```
newArr.push(fn.call(thisArg, this[i], i, this));
```